dotnet/src/Connectors/Connectors.Gemini/TextGeneration/GeminiTextGenerationService.cs

```diff
@@ -0,0 +1,121 @@
+        this._httpClient = HttpClientProvider.GetHttpClient(httpClient);
+        this._attributes.Add(AIServiceExtensions.ModelIdKey, this._model);
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyDictionary<string, object?> Attributes => this._attributes;
+
+    /// <inheritdoc />
+    public async Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        using var httpRequestMessage = this.VerifyArgumentsAndGetHTTPRequestMessage(prompt, executionSettings);
+
+        using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        var body = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+
+        var textGenerationResponse = JsonSerializer.Deserialize<TextGenerationResponse>(body);
+        if (textGenerationResponse is null)
+        {
+            throw new KernelException("Unexpected response from model")
+            {
+                Data = { { "ResponseData", body } },
+            };
+        }
+
+        return textGenerationResponse.Candidates.Select(c => new TextContent(c.Content.Parts[0].Text,
+            this.GetModelId(), textGenerationResponse)).ToList().AsReadOnly();
+    }
+
+    private HttpRequestMessage VerifyArgumentsAndGetHTTPRequestMessage(string prompt, PromptExecutionSettings? executionSettings)
+    {
+        Verify.NotNullOrWhiteSpace(prompt);
+        if (executionSettings is not GeminiPromptExecutionSettings geminiExecutionSettings)
+        {
+            throw new ArgumentException($"The execution settings must be of type {nameof(GeminiPromptExecutionSettings)}.");
+        }
+
+        var httpRequestMessage = this.GetHTTPRequestMessage(prompt, geminiExecutionSettings);
+        return httpRequestMessage;
+    }
+
+    private HttpRequestMessage GetHTTPRequestMessage(string prompt, GeminiPromptExecutionSettings geminiExecutionSettings)
+    {
+        var httpContent = GetHttpJsonContent(prompt, geminiExecutionSettings);
+        Uri uri = this.GetUriForGetText();
+        var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, uri);
+        httpRequestMessage.Content = httpContent;
+        httpRequestMessage.Headers.Add("User-Agent", HttpHeaderValues.UserAgent);
+        return httpRequestMessage;
+    }
+
+    private Uri GetUriForGetText() => new($"{GeminiApiEndpointBase}/{this._model}:generateContent?key={this._apiKey}");
+
+    private static ByteArrayContent GetHttpJsonContent(string prompt, GeminiPromptExecutionSettings geminiExecutionSettings)
+    {
+        var requestJsonObject = TextGenerationRequest.GenerateJsonFromPromptExecutionSettings(prompt, geminiExecutionSettings);
+        var requestUtf8Bytes = Encoding.UTF8.GetBytes(requestJsonObject.ToJsonString());
```

Comment:
You should be able to serialize directly to bytes rather than going through an intermediary string